packages/editor/src/client/atoms/emojiButton.tsx

```diff
@@ -0,0 +1,84 @@
+import React, {forwardRef, ButtonHTMLAttributes, useState, useRef} from 'react'
+import {useSlate} from 'slate-react'
+
+import {Icon, Popover, Whisper} from 'rsuite'
+import {SVGIcon} from 'rsuite/lib/@types/common'
+import {IconNames} from 'rsuite/lib/Icon/Icon'
+
+import './emojiButton.less'
+import {Picker, BaseEmoji} from 'emoji-mart'
+
+export interface EmojiButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  readonly icon: IconNames | SVGIcon
+  readonly iconActive?: IconNames | SVGIcon
+  readonly active?: boolean
+}
+
+export const BaseEmojiButton = forwardRef<HTMLButtonElement, EmojiButtonProps>(
+  function BaseEmojiButton({icon, iconActive, active, ...props}, ref) {
+    icon = iconActive && active ? iconActive : icon
+    return (
+      <button
+        style={{
+          border: active ? 'blue 1px solid' : '',
+          fontSize: 16,
+
+          cursor: 'pointer',
+          borderRadius: 3,
+          backgroundColor: 'transparent',
+
+          padding: 2
+        }}
+        ref={ref}
+        {...props}>
+        <Icon icon={icon} element={icon} />
+      </button>
+    )
+  }
+)
+
+export function EmojiButton({icon, iconActive}: EmojiButtonProps) {
+  const editor = useSlate()
+  const [isEmojiPickerOpen, setIsEmojiPickerOpen] = useState(false)
+
+  const triggerRef = useRef<any>(null)
+
+  const open = () => {
+    if (triggerRef.current) {
+      triggerRef.current!.open()
+      // TODO rather be done with forwardRef ?
```

Comment:
Is this TODO still open?

Reply:
Yes, I think it would be a bit cleaner with forwarding the ref to the BaseEmojiButton and using ref.current.isOpen (or similar) instead of using the isEmojiPickerOpen state variable to have the DOM element of single source of truth here. So maybe we could make a github <good first issue> out of it?

Reply:
@faebebin yes create a github issue out of this.

Reply:
ok